Test/find on shelf

Adamik10 opened the pull request:
#### Link to issue
https://reload.atlassian.net/browse/DDFSOEG-247

#### Description
This PR introduces a new story - periodical - for the find on shelf component, a cypress test for all the component stories (default + periodical), alongside a new fixture. 

#### Checklist

- [x] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [x] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [x] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

#### Additional comments or questions
n/a

Commit:
5411: use search result pager in loan list

Commit:
5411: use search result pager in loan list

Commit:
5411: use search result pager in loan list

Commit:
5411: create pagination component

Commit:
5411: create pagination component

Commit:
5411: add tests of loan list pagination

Commit:
5411: revert debugging

Commit:
5411: revert debugging

Commit:
5387: sort list with lodash instead

Commit:
5387: refactor to accomodate publizon

Commit:
5387: digital loans and refactoring

Commit:
5387: update mock publizon data

Commit:
5387: pagination and modals

Commit:
5387: add comment

Commit:
5387: revert publizon fetcher

Commit:
5387: remove renewable from loan list

Commit:
5387: revert fetchers

Commit:
5387: fixes from rebase

Commit:
5387: fixes from rebase

Commit:
5387: remove unnecessary div, rename string

Commit:
5387: remove lodash

Commit:
5387: rename

Commit:
5387: remove sortby import

Commit:
5387: remove unnecessary if

Commit:
5387: create hlper functions

Commit:
5387: lint

Commit:
5387: modal test edited

Commit:
5387: fix due date modal test

Commit:
5387: simplify modals

Commit:
5387: remove pagination and pagesize (rebase error)

Commit:
5387: revert rebase string change

Commit:
5387: fixes after rebase

Commit:
5387: fix import after rebase

Commit:
5387: fix import after rebase

Commit:
5379: reservation list files added

Commit:
5379: move stackable material

Commit:
5379: refactor before creating reservation materials

Commit:
5379:fix loantype materialtype error

Commit:
Merge branch 'release/4' into feature/5379-reservation-list-files-created

Commit:
Merge branch 'feature/5379-reservation-list-files-created' into feature/5379-refactor-before-reservation-material

Commit:
Merge branch 'feature/5379-reservation-list-files-created' into feature/5379-refactor-before-reservation-material

Commit:
Merge branch 'release/4' into feature/5379-refactor-before-reservation-material

Commit:
5379: fix build

Commit:
Merge branch 'release/4' into feature/5379-refactor-before-reservation-material

Commit:
5379: add getMaterialInfogetMaterialInfo

Commit:
5379: linting and remove empty file

Adamik10 committed:
Introduce a helper function to count total branches that have a material

Adamik10 committed:
Use totalBranchesHaveMaterial() in Find on shelf modal

To accurately show how many branches have a specific material.

Adamik10 committed:
Translate all non-English system texts for search header app to English

Adamik10 committed:
Translate all non-English system texts in search result app to English

Adamik10 committed:
Translate all non-English system texts in material app to English

Adamik10 committed:
Correct material app test to match the new system texts

Adamik10 committed:
Correct search header app test to follow new system texts

Adamik10 committed:
Correct capitalization of "In series" system text to "in series"

Adamik10 committed:
Correct search result app test to follow updated system texts

Commit:
add branches data to fbs

Adamik10 committed:
Merge pull request #176 from reload/fix/find-on-shelf-x-branches-have-the-material

Fix/find on shelf x branches have the material

Adamik10 committed:
Merge branch 'release/4' into fix/all-system-texts-in-english

Adamik10 committed:
Correct material app periodical test to follow new system texts

Adamik10 committed:
Merge pull request #177 from reload/fix/all-system-texts-in-english

Fix/all system texts in english

sinejespersen committed:
add newline

Commit:
Merge pull request #175 from itk-dev/feature/5379-refactor-before-reservation-material

Feature/5379 refactor before reservation material

Commit:
Merge pull request #178 from itk-dev/feature/branches-json

add branches data to fbs

Commit:
5387: wiremock files cors fix

Commit:
5387: remove verbose

Commit:
Merge pull request #182 from itk-dev/feature/5387-wiremock-data

5387: wiremock files cors fix

Adamik10 committed:
Introduce a mocked data fixture for find on shelf holdings

This will be used for testing purposes.

Adamik10 committed:
Introduce mocked manifestation data for find on shelf periodical story

Adamik10 committed:
Introduce "Find on shelf modal - periodical" story

Adamik10 committed:
Introduce cypress test for find on shelf stories

Adamik10 committed:
Trim down find on shelf holdings fixture to include relevant data only

The fixture was 18k lines of unnecessary data.

Adamik10 committed:
Update find on shelf test to match its trimmed down fixture

Adamik10 committed:
Test find on shelf for whether the main branch shows up first

kasperg commented on src/components/find-on-shelf/find-on-shelf.test.ts:
```diff
@@ -0,0 +1,70 @@
+      })
+      .as("Find on shelf availability");
+    cy.visit("/iframe.html?id=components-find-on-shelf-modal--periodical");
+    cy.contains("button:visible", "FIND PÅ HYLDEN").click();
```
Consider whether the text "FIND PÅ HYLDEN" needs to be in all caps. In general I would expect such stylistic casing to be controlled with CSS.

Consider whether this text "FIND PÅ HYLDEN" should be in English as it has been decided that this is the default language - also for controls in Storybook.

Adamik10 replied:
Great points! Fixed both ;)

kasperg commented on src/components/find-on-shelf/find-on-shelf.test.ts:
```diff
@@ -0,0 +1,70 @@
+    cy.get("select").contains("29");
+  });
+
+  it("Doesn't contain two dropdowns when it's not a periodical", () => {
+    cy.visit("/iframe.html?id=components-find-on-shelf-modal--default");
+    cy.contains("button:visible", "FIND PÅ HYLDEN").click();
+    cy.get("select").should("not.exist");
+  });
```
Please create different tests for different cases: 1) Find on self default 2) Find on shelf for periodicals

Adamik10 replied:
Very good thinking!

kasperg commented on src/components/find-on-shelf/find-on-shelf.test.ts:
```diff
@@ -0,0 +1,70 @@
+
+  it("Contains text with the sum of all available branches", () => {
+    cy.contains("biblioteker har materialet");
+  });
```
Please include the actual number in the test.

This is the closest we can get to testing the counting logic.

Also note the language.

kasperg commented on src/components/find-on-shelf/find-on-shelf.test.ts:
```diff
@@ -0,0 +1,70 @@
+  });
+
+  it("Contains a summary for the main branch with no available speciments", () => {
+    cy.get("summary").contains("Hovedbiblioteket");
+    cy.get("summary").contains("unavailable");
```
```suggestion
    cy.get("summary")
      .contains("Hovedbiblioteket")
      .parent()
      .contains("unavailable");
```

Consider whether you need to chain the `contains()` assertions to check whether the item is actually unavailable at the main library branch or just unavailable at some library branch.

Adamik10 replied:
That is very correct. Mine was just checking any for unavailable. 
🥇 🏆

kasperg reviewed (changes requested):
:+1: Good stuff here. I have a few things for you to look into.

In general the whole `selectedPeriodical` arg with control type null etc. seems a bit odd. Perhaps it is just me who has not seen this pattern before.

Adamik10 committed:
Merge pull request #174 from reload/feature/find-on-shelf/periodical-with-release4

Feature/find on shelf periodical dropdowns

Adamik10 committed:
Use english system texts in find on shelf modal tests

Adamik10 committed:
Separate find on shelf test to describe find on shelf stories separately

Adamik10 committed:
Look for Main library branch with unavailable lable in FindOnShelf test

Before we just looked for main branch, and then looked for unavailable label (on any branch) which was incorrect.

Adamik10 commented:
To follow this PR, please go to PR #187 in the upstream fork
